testsuite/MDAnalysisTests/core/util.py

```diff
@@ -438,7 +440,29 @@ def unwrapped_coords(self, compound, reference):
                                              [0.44, 0.65, 0.6]],
                                             dtype=np.float32)
 
-        # apply y- and z-dependent shift of x and y coords for triclinic boxes:
+            if reference_point is not None and compound is not "group":
+                for base in range(3):
+                    loc_centre = np.mean(relpos[base, :], axis=0)
+                    shift = np.rint((reference_point - loc_centre) / self._box_edge)
+                    relpos[base, :] += shift
+
+                for base in range(3, 15, 3):
+                    loc_centre = np.mean(relpos[base:base + 3, :], axis=0)
+                    shift = np.rint((reference_point - loc_centre) / self._box_edge)
+                    relpos[base:base + 3, :] += shift
+
+                # type C
+                for base in range(15, 23, 4):
+                    loc_centre = np.mean(relpos[base:base + 4, :], axis=0)
+                    shift = np.rint((reference_point - loc_centre) /self._box_edge)
+                    relpos[base:base + 4, :] += shift
+
+                # type D
+                for base in range(23, 47, 8):
+                    loc_centre = np.mean(relpos[base:base + 8, :], axis=0)
+                    shift = np.rint((reference_point - loc_centre) / self._box_edge)
+                    relpos[base:base + 8, :] += shift
+
         if self._is_triclinic:
```

Comment:
@richardjgowers , @jbarnoud I am stuck at writing tests for triclinic systems. Does this **if block** convert the coordinates of a molecule from orthorhombic system to triclinic system?

Reply:
Indeed. From what I understand, the coordinates of each molecule is provided in an orthorhombic system and are shifted in the case of a non-orthorhombic system (orthorhombic is a special case of triclinic, so in all cases it is triclinic) so that they still match their description.

Reply:
@NinadBhat I designed the UnWrapUniverse so that it first initializes all coordinates in an orthorhombic box. Thereafter, if the box is actually supposed to be triclinic, it applies a position-dependent shift (the "skew" of the box) to the x- and y-components of all coordinates. That's what's being done here.

Reply:
Think of it as taking an orthorhombic box and then shearing the top face in the x- and y-direction without altering the height of the box.